### ISSUE TITLE
Error for convolutions with non-default stride/padding

### ISSUE THREAD

**Issue by TTitcombe** [facebookresearch/pytorch-dp#1]
Reshape in `autograd_grad_sample.compute_grad_sample` fails for a convolution layer with non-default parameters.

It's caused by the shapes in `A` and `B` not matching up. I think the fix is a case of changing `A` to

```
A = torch.nn.functional.unfold(A, layer.kernel_size, padding=layer.padding, stride=layer.stride)
```

**Comment by Darktex** [comment]
Hi @TTitcombe ! Indeed this is a bug. Thank you for reporting it! Investigating it this week :)

**Comment by jmg-74** [comment]
Not related with the subject, **sorry** ! Bad copy-paste manip. :(
